content/docs/reference-react-component.md

```diff
@@ -476,73 +477,73 @@ React doesn't call `UNSAFE_componentWillReceiveProps()` with initial props durin
+## API-ים אחרים {#other-apis-1}
 
-Unlike the lifecycle methods above (which React calls for you), the methods below are the methods *you* can call from your components.
+לא כמו מתודות מעגל-החיים למעלה (אשר React קוראת להן עבורנו), המתודות להלן הן מתודות *שאנחנו* יכולים לקרוא להן מהקומפוננטות שאנו יוצרים.
 
-There are just two of them: `setState()` and `forceUpdate()`.
+יש רק שניים כאלו: `setState()` ו- `forceUpdate()`.
 
 ### `setState()` {#setstate}
 
 ```javascript
 setState(updater[, callback])
 ```
 
-`setState()` enqueues changes to the component state and tells React that this component and its children need to be re-rendered with the updated state. This is the primary method you use to update the user interface in response to event handlers and server responses.
+`setState()` מכניסה לתור שינויים ב-state של הקומפוננטה ואומרת ל-React שהקומפוננטה הזו והילדים שלה צריכים להיות מרונדרים מחדש עם ה-state המעודכן. זו המתודה הראשית שבה משתמשים כדי לעדכן את ממשק המשתמש בתגובה למנהלי אירועים ותגובות שרת.
 
-Think of `setState()` as a *request* rather than an immediate command to update the component. For better perceived performance, React may delay it, and then update several components in a single pass. React does not guarantee that the state changes are applied immediately.
+חישבו על `setState()` כאל *בקשה* מאשר כאל פקודה מידית לעדכון הקומפוננטה. למען ביצועים טובים יותר, React עשויה לעכב את ביצועה של הפקודה, ואז לעדכן מספר קומפוננטות במעבר אחד. React לא מבטיחה ששינוי ה-state יתבצעו מידית.
 
-`setState()` does not always immediately update the component. It may batch or defer the update until later. This makes reading `this.state` right after calling `setState()` a potential pitfall. Instead, use `componentDidUpdate` or a `setState` callback (`setState(updater, callback)`), either of which are guaranteed to fire after the update has been applied. If you need to set the state based on the previous state, read about the `updater` argument below.
+`setState()` לא תמיד מעדכנת באופן מידי את הקומפוננטה. היא עשויה לדחות את העידכון לאחר כך. זה גורם לכך שקריאה מ- `this.state` ישר אחרי קריאה ל-`setState()` למלכודת פוטנציאלית. במקום, השתמשו ב- `componentDidUpdate` או בפונקצית ה-callback ב-`setState` (`setState(updater, callback)`), בכל אחד מהמקרים מובטח כי יתבצע לאחד שהעדכון יקרה. אם יש צורך להגדיר את ה-state בהתבסס על ה-state הקודם, קראו על הארגומנט `updater` מטה.
 
-`setState()` will always lead to a re-render unless `shouldComponentUpdate()` returns `false`. If mutable objects are being used and conditional rendering logic cannot be implemented in `shouldComponentUpdate()`, calling `setState()` only when the new state differs from the previous state will avoid unnecessary re-renders.
+`setState()` תמיד יוביל לרינדור מחדש אלא אם `shouldComponentUpdate()` תחזיר `ביטוי שקר`. אם אוביקטי מיוטבל נמצאים בשימוש ולוגיקת רינדור מותנה לא יכולה להיות ממומשת ב- `shouldComponentUpdate()`, קריאה ל- `setState()` רק כאשר ה-state החדש שונה מהקודם תמנע רינדור מחדש מיותר.
```

Comment:
```suggestion
`setState()` תמיד יוביל לרינדור מחדש אלא אם `shouldComponentUpdate()` תחזיר `ביטוי שקר`. אם אובייקטים שניתנים לשינוי נמצאים בשימוש ולוגיקת רינדור מותנה לא יכולה להיות ממומשת ב- `shouldComponentUpdate()`, קריאה ל- `setState()` רק כאשר ה-state החדש שונה מהקודם תמנע רינדור מחדש מיותר.
```

See my suggestion, because אוביקטי מיוטבל sounds wrong to me :/

Reply:
yeah, it was a tricky one. I didn't really like it either, but on the other hand also "שניתנים לשינוי" doesn't make enough sense to me. because mutable is the name of that term alongside to its actual meaning, in my opinion we should keep at least a reference to that. such as "שניתנים לשינוי (mutable)"

Reply:
@galtalmor can you help out here?

Reply:
@itzikya Now in a look back I think your suggestion "שניתנים לשינוי (mutable)" is good enough, can you commit it?

Reply:
you got it